src/librustdoc/scrape_examples.rs

```diff
@@ -162,19 +200,28 @@ crate fn run(
     renderopts: config::RenderOptions,
     cache: formats::cache::Cache,
     tcx: TyCtxt<'_>,
-    example_path: PathBuf,
+    options: ScrapeExamplesOptions,
 ) -> interface::Result<()> {
     let inner = move || -> Result<(), String> {
         // Generates source files for examples
         let (cx, _) = Context::init(krate, renderopts, cache, tcx).map_err(|e| e.to_string())?;
 
         // Run call-finder on all items
+        let find_crate = |target_crate: String| {
+            tcx.crates(())
+                .iter()
+                .find(|crate_num| tcx.crate_name(**crate_num).as_str() == target_crate.as_str())
```

Comment:
Note that if two different versions of the crate are in the dependency tree this will collect examples from both. That seems like the right behavior, but can you leave a comment so it's more clear?

Reply:
Oh err hmm I misread the code before - this uses `find`, not filter. So it will only use the first crate with the matching name. Can you change it to use filter instead?

Reply:
Ah whoops fixed